CommonCompilerConfig.cmake

```diff
@@ -220,6 +229,12 @@ elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
   set(shared_linker_flags_unoptimized "${shared_linker_flags_unoptimized} -Wl,-undefined,error")
 endif()
 
+# Compatibility with gcc >= 4.9 which requires the use of gcc's own wrappers for ar and ranlib in combination with LTO
+# works also with LTO disabled
+IF(UNIX AND CMAKE_COMPILER_IS_GNUCXX AND (NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug") AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "4.9" OR CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "4.9"))
```

Comment:
To avoid duplication of logic, let's set `CMAKE_AR` and `CMAKE_RANLIB` here as well along with a short explanatory comment to the effect that `COMMON_STATICLIB_CMAKE_DEFINITIONS_OPTIMIZED` is a workaround for a cmake bug (including a link to https://gitlab.kitware.com/cmake/cmake/issues/15547), instead of doing it separately in the root CMakeLists.txt. Also, document `CMAKE_AR` and `CMAKE_RANLIB` alongside [CMAKE_LINKER](https://github.com/hstm/nupic.core/blob/8b6caad44b3593e2c29b91c194cc619114eb9c6a/CommonCompilerConfig.cmake#L64-L65)